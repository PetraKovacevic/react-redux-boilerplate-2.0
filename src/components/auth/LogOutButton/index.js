import React from 'react';
import { connect } from 'react-redux';
import { Button } from 'react-bootstrap';
import { bool } from 'prop-types';

import { unauthenticate } from '@/services/session/actions';
import { redirect } from '@/services/redirect';
import { redirects } from '@/config';
import * as userStorage from '@/data/users/services/storage';

const signOut = props => {
    props.unauthenticate();
    userStorage.removeCurrentUser();
    redirect(redirects.unathenticated);
};

export const LogOutButton = props => {
    return props.authenticated ? <Button onTouchTap={() => { signOut(props); }} >Log Out</Button> : null;
};

LogOutButton.propTypes = {
    authenticated: bool
};

const mapStateToProps = (state) => {
    return {
        authenticated: state.services.session.authenticated
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        unauthenticate: () => {
            dispatch(unauthenticate());
        }
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(LogOutButton);
