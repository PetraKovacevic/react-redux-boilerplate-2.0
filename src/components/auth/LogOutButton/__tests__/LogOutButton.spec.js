import React from 'react';
import { shallow, mount, render } from 'enzyme';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

import LogOutButton from '../index';

describe('Log out button page', function () {
    describe('Is logged in', function () {
        const mockStore = configureStore();
        let store, container;
        beforeEach(() => {
            store = mockStore({
                services: {
                    session: {
                        authenticated: true
                    }
                }
            });
            container = shallow(<LogOutButton store={store} />);
        });

        it('should render without throwing an error', function () {
            expect(container.length).toEqual(1);
        });

        it('should render button if authenticated is true', function () {
            const tree = renderer.create(
                <LogOutButton store={store} />
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });

    });

    describe('Is NOT logged in', function () {
        const mockStore = configureStore();
        let store, container;

        beforeEach(() => {
            store = mockStore({
                services: {
                    session: {
                        authenticated: false
                    }
                }
            });
            container = shallow(<LogOutButton store={store} />);
        });

        it('should return null if authenticated is false', function () {
            expect(container.html()).toEqual('');
        });

    });

});
