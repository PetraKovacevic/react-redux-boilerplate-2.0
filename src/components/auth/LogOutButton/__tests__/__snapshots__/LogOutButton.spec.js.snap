// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Log out button page Is logged in should render button if authenticated is true 1`] = `
<button
  className="btn btn-default"
  disabled={false}
  onTouchTap={[Function]}
  type="button"
>
  Log Out
</button>
`;
